include/odp/api/spec/ipsec.h

```diff
@@ -739,6 +739,36 @@ typedef struct odp_ipsec_sa_param_t {
 			 */
 			odp_cos_t dest_cos;
 
+			/** Enable reassembly of pre-IPsec IP fragments
+			 *
+			 *  When enabled, pktio input will attempt reassembly of
+			 *  pre-IPsec fragmented IP packets. This reassembly
+			 *  attempt would be performed following IPsec inbound
+			 *  processing.
+			 *
+			 *  The pktio input processing IPsec should have
+			 *  'ip_reassem' enabled to have this feature supported.
+			 *
+			 *  @see odp_pktio_config(), odp_reassembly_config()
+			 *
+			 */
+			odp_bool_t pre_frag_ip_reassem;
+
+			/** Enable reassembly of post-IPsec IP fragments
+			 *
+			 *  When enabled, pktio input will attempt reassembly of
+			 *  post-IPsec fragmented IP packets. This reassembly
+			 *  attempt would be performed prior to IPsec inbound
+			 *  processing.
+			 *
+			 *  The pktio input processing IPsec should have
+			 *  'ip_reassem' enabled to have this feature supported.
+			 *
+			 *  @see odp_pktio_config(), odp_reassembly_config()
+			 *
+			 */
+			odp_bool_t post_frag_ip_reassem;
```

Comment:
Why is this needed? Why is the setting is pktio config enough to cause reassembly of all incoming packets, whether they get IPsec processed or not? Isn't this thus somewhat redundant?

Reply:
One thought process is treating these different HW/ODP object and it can be shared with different odp_packe_t. 
ie. create, odp_reassem_t object with configuration parameter and the same odp_reassem_t can be shared with multiple pktio as the configuration is per reassembly HW not for the packet_io.

Reply:
> 
> 
> Why is this needed? Why is the setting is pktio config enough to cause reassembly of all incoming packets, whether they get IPsec processed or not? Isn't this thus somewhat redundant?

Intention was to differentiate both cases of reassembly possible with IPsec. Agreed that per SA control of post_frag reassembly is not possible. Will remove this option as part of V3.